### PR TITLE
feat: use updateMySocials not updateUserFlag

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

After https://github.com/freeCodeCamp/freeCodeCamp/pull/46378 goes in, I'll update this to remove any reference to `updateUserFlag`

### CODE REVIEW

**Comment by raisedadead** [comment]
Yo! This has been sitting here. Do you want to give this a quick rebase and see if we can add this to the review queue?

**Comment by ojeytonwilliams** [comment]
This now removes the last of `updateUserFlag` from the client.  After this is deployed we can finally remove the endpoint.

Sorry for the delay, Mrugesh!

**Review by naomi-lgbt — DISMISSED** [review]
Should be good to go.

File: client/src/components/settings/internet.tsx [hunk 1]
```diff
@@ -142,15 +141,9 @@ class InternetSettings extends Component<InternetProps, InternetState> {
...
-        isGithub: !!formValues.githubProfile,
-        isLinkedIn: !!formValues.linkedin,
-        isTwitter: !!formValues.twitter,
-        isWebsite: !!formValues.website
-      };
 
-      const { updateInternetSettings } = this.props;
-      return updateInternetSettings({ ...isSocial, ...formValues });
+      const { updateSocials } = this.props;
+      return updateSocials({ ...formValues });
```
Comment (ShaunSHamilton) [inline client/src/components/settings/internet.tsx:146]:
```suggestion
      const isSocial = {
        isGithub: !!formValues.githubProfile,
        isLinkedIn: !!formValues.linkedin,
        isTwitter: !!formValues.twitter,
        isWebsite: !!formValues.website
      };
      return updateSocials({ ...isSocial, ...formValues });
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
We need this, otherwise the social icons do not appear.

Perhaps it would be best changing the logic in `social-icons.tsx`, though.

File: client/src/components/settings/internet.tsx (hunk 1 above)
Comment (naomi-lgbt) [reply client/src/components/settings/internet.tsx:146]: Yeah, I'd be in favour of removing those booleans. It doesn't make sense to have a boolean "github exists" value and the actual github link lol

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

**Comment by ojeytonwilliams** [comment]
The api is fine with these changes, but should be updated after we've deployed.  Otherwise it'll keep dutifully creating `isGithub` and so on, only to be ignored by the client.

**Review by ShaunSHamilton — APPROVED** [review]
Th removal of a saga is a good day 👍

**Comment by ahmadabdolsaheb** [comment]
It seems like we are not doing much when we execute updateMyThemeError, but that is for another pr.

**Review by ahmaxed — APPROVED** [review]

**Pull request merged**